client/src/pages/course/schedule.tsx

```diff
@@ -34,23 +39,49 @@ export function SchedulePage(props: CoursePageProps) {
     [courseService],
   );
 
+  let schedule;
+
+  switch (scheduleViewMode) {
+    case 'Table':
+      schedule = <TableView data={data} timeZone={timeZone} />;
+      break;
+    case 'List':
+      schedule = <ListView data={data} timeZone={timeZone} />;
+      break;
+    case 'Calendar':
+      schedule = <CalendarView data={data} timeZone={timeZone} />;
+      break;
+    default:
+      schedule = <TableView data={data} timeZone={timeZone} />;
+      break;
+  }
```

Comment:
WDYT about simplifying this part? For example replace the switch with the map:

```
const mapScheduleViewToComponent = {
  table: TableVie,
  tist: ListView,
  calendar: CalendarView,
};
``` 
and later:
```
const ScheduleView = mapScheduleViewToComponent[scheduleViewMode] || TableView;
```
next:
```
<ScheduleView data={data} timeZone={timeZone} />
```

Reply:
done